api/src/main/java/com/datatorrent/api/Operator.java

```diff
@@ -172,6 +172,51 @@ public String toString()
 
   }
 
+  public interface ProxyPort<T> extends Port
+  {
+    void set(T port);
+
+    T get();
+  }
+
+  public final class ProxyInputPort<T> extends DefaultInputPort<T> implements ProxyPort<InputPort<T>>
```

Comment:
ProxyInputPort does not behave the same way as DefaultInputPort. Implementation of InputPort interface in ProxyInputPort should be different compared to DefaultInputPort implementations. Most of the method should either throw an exception or delegate to internal module port connected to it.